blocks/subscriptions-block/components/useOffer.jsx

```diff
@@ -0,0 +1,54 @@
+import { useState, useEffect, useCallback } from 'react';
+import { useFusionContext } from 'fusion:context';
+import getProperties from 'fusion:properties';
+import { isServerSide } from '@wpmedia/engine-theme-sdk';
+
+const offerService = ({
+  origin,
+  code,
+  endpoint,
+}) => fetch(`${origin}${endpoint}${code || 'default'}`, {})
+  .then((res) => res.json());
+
+const useOffer = ({ campaignCode }) => {
+  const { arcSite } = useFusionContext();
+  const { api: { retail: { origin = null, endpoint } } } = getProperties(arcSite);
```

Comment:
You can remove the null default here and it will still work (also give 100% test coverage with the additional test I include below).... but it points me to a possible different issue.  We are just running the fetch with whatever is passed instead of checking the values are there and erroring out.